html/captive-portal/lib/captiveportal/Base/Controller.pm

```diff
@@ -56,12 +56,9 @@ sub reached_retry_limit {
     return 0 unless $max;
     my $cache = $c->user_cache;
     my $retries = $cache->get($retry_key) || 1;
-    if($retries > $max ) {
-        return 1;
-    }
     $retries++;
-    $cache->set($retry_key,$retries);
-    return 0;
+    $cache->set($retry_key,$retries,$c->profile->{_block_interval});
```

Comment:
If the other file is pf::config in the readProfileConfigFile function.
I am massaging the data from conf/profile.conf not accessing the data from a pf::Portal::Profile object.
However this miss match can cause confusion.
We should move the pf::Portal::Profile to a Moo/Moose object to avoid this confusion in the future.